repo2docker/apt.txt

```diff
@@ -0,0 +1,71 @@
+# common cli tools and helpers
+bash-completion
+nano
+vim
+
+# installing octave here because if we do it through conda, it is going to look for a specific compiler that no longer exists when we do `pkg install` anything.
+# sh: 1: /home/conda/feedstock_root/build_artifacts/octave_1587137750417/_build_env/bin/x86_64-conda_cos6-linux-gnu-c++: not found
+# relevant issue: https://github.com/ContinuumIO/anaconda-issues/issues/9863 even if it's a different language
+# octave_kernel still installed through conda, though
+octave
+octave-*
+
+# install all dependencies of texlive-full except for...
+# latex-cjk-all, texlive-fonts-extra, texlive-fonts-extra-doc, texlive-fonts-extra-links and all texlive-lang-* except texlive-lang-english
+# these were excluded to reduce image size and we don't expect high usage for those dependencies
+
+biber
+chktex
+cm-super
+context
+dvidvi
+dvipng
+feynmf
+fragmaster
+info
+lacheck
+latexdiff
+latexmk
+lcdf-typetools
+lmodern
+prerex
+psutils
+purifyeps
+t1utils
+tex-gyre
+texinfo
+texlive-base
+texlive-bibtex-extra
+texlive-binaries
+texlive-extra-utils
+texlive-font-utils
+texlive-fonts-recommended
+texlive-fonts-recommended-doc
+texlive-formats-extra
+texlive-games
+texlive-humanities
+texlive-humanities-doc
+texlive-lang-english
+texlive-latex-base
+texlive-latex-base-doc
+texlive-latex-extra
+texlive-latex-extra-doc
+texlive-latex-recommended
+texlive-latex-recommended-doc
+texlive-luatex
+texlive-metapost
+texlive-metapost-doc
+texlive-music
+texlive-pictures
+texlive-pictures-doc
+texlive-plain-generic
+texlive-pstricks
+texlive-pstricks-doc
+texlive-publishers
+texlive-publishers-doc
+texlive-science
+texlive-science-doc
+texlive-xetex
+tipa
+vprerex
+xindy
```

Comment:
Here are some basic tools that I think we should include:

```
build-essential
clang
cmake
curl
ffmpeg
gfortran
git
htop
mercurial
subversion
nodejs
pandoc
tree
```

Or are there a set of default utils installed already?

Reply:
Added clang, cmake, ffmpeg, htop, pandoc and tree. Everything else listed there is already available.